src/api/client/Claims.ts

```diff
@@ -483,4 +486,50 @@ export class Claims {
    *  - a custom claim type with the same `name` already exists
    */
   public registerCustomClaimType: ProcedureMethod<RegisterCustomClaimTypeParams, BigNumber>;
+
+  /**
+   * Retrieves a custom claim type based on its name
+   *
+   * @param name - The name of the custom claim type to retrieve
+   */
+  public async getCustomClaimTypeByName(name: string): Promise<CustomClaimType | null> {
+    const {
+      context: {
+        polymeshApi: {
+          query: { identity },
+        },
+      },
+    } = this;
+
+    const customClaimTypeIdOpt = await identity.customClaimsInverse(name);
+
+    if (customClaimTypeIdOpt.isEmpty) {
+      return null;
+    }
+
+    return { id: u32ToBigNumber(customClaimTypeIdOpt.value), name };
+  }
+
+  /**
+   * Retrieves a custom claim type based on its ID
+   *
+   * @param id - The ID of the custom claim type to retrieve
+   */
+  public async getCustomClaimTypeById(id: BigNumber): Promise<CustomClaimType | null> {
+    const {
+      context: {
+        polymeshApi: {
+          query: { identity },
+        },
+      },
+    } = this;
+
+    const customClaimTypeIdOpt = await identity.customClaims(bigNumberToU32(id, this.context));
```

Comment:
`this.context` can also be destructed above and directly `context` can be used here